ursa-major-schema.json

```diff
@@ -0,0 +1,284 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"$id": "https://ursa-major-schema.rockarch.org/",
+	"type": "object",
+	"title": "Ursa Major Schema",
+	"oneOf": [{
+			"$ref": "#/definitions/accession"
+		},
+		{
+			"$ref": "#/definitions/bag"
+		}
+	],
+	"definitions": {
+		"accession": {
+			"$id": "#/definitions/accession",
+			"type": "object",
+			"title": "Accession",
+			"required": ["title"],
+			"properties": {
+				"count": {
+					"type": "integer",
+					"example": 123
+				},
+				"next": {
+					"type": ["string", "null"]
+				},
+				"previous": {
+					"type": ["string", "null"]
+				},
+				"results": {
+					"type": "array",
+					"items": {
+						"properties": {
+							"url": {
+								"type": "string"
+							},
+							"title": {
+								"type": "string",
+								"maxLength": 256
+							},
+							"accession_number": {
+								"type": ["string", "null"],
+								"maxLength": 10
+							},
+							"created": {
+								"type": "string",
+								"format": "date-time"
+							},
+							"last_modified": {
+								"type": "string",
+								"format": "date-time"
+							}
+						}
+					}
+				}
+			}
+		},
+		"bag": {
+			"$id": "#/definitions/bag",
+			"type": "object",
+			"title": "Bag",
+			"required": ["organization", "bag_it_name", "identifier", "origin"],
```

Comment:
I think "organization" will be an issue for bags that have not come from Aurora. I think there are 2 possibilities:

1. Have "organization" be a requirement for digitized and legacy digital processing content (i.e., content that is posted to Ursa Major from an application other than Aurora).
2. Not make "organization" be required in the schema.

I don't love option #1, so would prefer the second solution. What do you think @helrond ?

Reply:
AFAIK there's no way to conditionally require things in JSONSchema, so we'd be talking about creating an additional schema, which I don't love. So I think I agree we should just make organization non-required.

Reply:
(I should clarify that by the first option I meant we need to have the Aurora-parallel app include "organization", but then we'd need to define what "organization" means for those workflows. Which opens a whole can of 🐛.)

Reply:
One potential caveat is that this might cause some issues down the line in Aquarius, because that app uses the `organization` field data in transformations. It's not an issue for digitized/legacy born digital because they bypass those services, but it could be an issue in the future if we plug other things into this infrastructure.

Reply:
By "source" you mean the originating application (e.g., Aurora) and not the workflow (e.g., digitized or legacy born digital)? I think that makes sense; while it's more complex I think it allows it to be more extensible.

Reply:
I think it depends a bit on where the validation sits. One option is to generalize it a bit and have it sit in Zodiac as part of the request handling in the gateway. The other is to keep it specific to the service, in which case we could tie it to "origin" rather than "source" per-se.